cli/src/offline.rs

```diff
@@ -22,6 +25,49 @@ pub const SIGNER_ARG: ArgConstant<'static> = ArgConstant {
     help: "Provid a public-key/signature pair for the transaction",
 };
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum BlockhashSpec {
```

Comment:
Seems like `BlockhashSpec` is pretty coupled to `sign_only` now. What if this was instead something like:

```rust
enum TxMode {
  SignOnly,
  Send(Option<Hash>),
}
```

Basically combining the `blockhash` and `sign_only` arguments together

Reply:
I feel like this somewhat over-encodes for the current state of the logic, which I'd like to move towards untangling.  My intent with the above is purely to intermediate the `rpc_client.get_recent_blockhash()` calls with user-supplied data.

Reply:
Ah I see, I've been pushing for encoding the expected execution paths. Seems that we have two outcomes for a lot of our cli commands now: an offline signed transaction and an online signed and sent transaction. Wouldn't that make the code more readable and tests easier to write and understand?

Reply:
I'd like to work towards any combination of present and absentee signers, their online-ness irrelevant, and anyone who happens to have all of the signatures for a transaction may submit it to the network

Reply:
Ah cool, nice to have that context. Works for me then!